### PR TITLE
Rust Conference Overview 2025

### PR DESCRIPTION
**Pull request by mre**

### CODE REVIEW

File: content/blog/rust-conferences-2025/index.md [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+- **Pricing**:
+  - Standard: $172 USD
+  - Student: $12 USD
+- **Includes**: Conference access, networking, beer meeting, afterparty
+- **Links**: [Website](https://www.rustikon.dev/) | [Tickets](https://www.rustikon.dev/tickets) | [Announcement](https://softwaremill.com/why-attend-a-rust-conference-for-developers-rustikon/)
+- **Social**: [@rustikonconf](https://x.com/rustikonconf) | [Mastodon](https://mastodon.social/@Rustikon)
+
+### RustAsia (Hong Kong)
+
+Another first-time event, RustAsia, is a conference for Rust developers in Asia.
```
Comment (M3t0r) [inline content/blog/rust-conferences-2025/index.md]:
```suggestion
Another first-time event, RustAsia is a conference for Rust developers in Asia.
```

File: content/blog/rust-conferences-2025/index.md [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+
+- **When**: March 26, 2025
+- **Where**: [POLIN Conference Centre, Warsaw](https://g.co/kgs/vL3DNky)
+- **Format**: Single-day conference
+- **Pricing**:
+  - Standard: $172 USD
+  - Student: $12 USD
+- **Includes**: Conference access, networking, beer meeting, afterparty
+- **Links**: [Website](https://www.rustikon.dev/) | [Tickets](https://www.rustikon.dev/tickets) | [Announcement](https://softwaremill.com/why-attend-a-rust-conference-for-developers-rustikon/)
+- **Social**: [@rustikonconf](https://x.com/rustikonconf) | [Mastodon](https://mastodon.social/@Rustikon)
```
Comment (M3t0r) [inline content/blog/rust-conferences-2025/index.md]: If "Mastodon" is Mastodon, then what is "@rustikonconf"?

File: content/blog/rust-conferences-2025/index.md [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+As per tradition, they kick off the year of Rust conferences. 
+
+- **When**: February 19-20, 2025
+- **Where**: [The Brewery, 52 Chiswell Street, London, EC1Y 4SD, UK](https://maps.app.goo.gl/QakezC3gjkyo7sNr9)
+- **Format**: In-person workshops + conference
+- **Pricing**: 
+  - Conference Only: 304 dollars (early bird)
+  - Workshops starting at $300 + VAT
+- **CFP**: Closed
+- **Links**: [Website](https://www.rustnationuk.com/) | [Schedule](https://www.rustnationuk.com/schedule) | [Past Talks (Videos)](https://www.youtube.com/channel/UCLksRXfBiEITZMUo2ssjSdA)
```
Comment (M3t0r) [inline content/blog/rust-conferences-2025/index.md:66]: No socials?

File: content/blog/rust-conferences-2025/index.md [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+  - Standard: $172 USD
+  - Student: $12 USD
+- **Includes**: Conference access, networking, beer meeting, afterparty
+- **Links**: [Website](https://www.rustikon.dev/) | [Tickets](https://www.rustikon.dev/tickets) | [Announcement](https://softwaremill.com/why-attend-a-rust-conference-for-developers-rustikon/)
+- **Social**: [@rustikonconf](https://x.com/rustikonconf) | [Mastodon](https://mastodon.social/@Rustikon)
+
+### RustAsia (Hong Kong)
+
+Another first-time event, RustAsia, is a conference for Rust developers in Asia.
+Allen Wyma, one of the conference organizers, is known for being the host of the Rustacean Station podcast.
```
Comment (M3t0r) [inline content/blog/rust-conferences-2025/index.md]: "the host": I thought they have different hosts and you can publish your own episodes? Sounds like a bullshit accolade.

File: content/blog/rust-conferences-2025/index.md [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+- **Links**: [Website](https://www.rustnationuk.com/) | [Schedule](https://www.rustnationuk.com/schedule) | [Past Talks (Videos)](https://www.youtube.com/channel/UCLksRXfBiEITZMUo2ssjSdA)
+
+### Rustikon (Warsaw, Poland)
+
+This is the first Rust conference in Poland.
+It aims to bring together Rust developers from the region and beyond.
+Expect a developer-friendly atmosphere with ~15 expert talks in a single-track format, perfect for staying connected and engaged. After a day of learning, unwind with us in a local pub and network with fellow devs. 
+
+- **When**: March 26, 2025
+- **Where**: [POLIN Conference Centre, Warsaw](https://g.co/kgs/vL3DNky)
```
Comment (M3t0r) [inline content/blog/rust-conferences-2025/index.md]: No Google Maps link like the other venues?

File: content/blog/rust-conferences-2025/index.md [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+- **CFP**: Open until December 31, 2024
+- **Links**: [Website](https://rustasiaconf.com/) | [CFP](https://rustasiaconf.com/proposals)
+
+## Q2 2025
+
+### RustWeek (Utrecht, Netherlands)
+
+RustWeek is a week-long event that combines talks, workshops, and social events.
+It's located in cozy Utrecht, the fourth-largest city in the Netherlands.
+There is a focus on Rust usage in the industry.
```
Comment (M3t0r) [inline content/blog/rust-conferences-2025/index.md]: Maybe mention it was known as RustNL before 2025?

File: content/blog/rust-conferences-2025/index.md [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+It's a great conference for Rust developers of all levels.
+As per tradition, they kick off the year of Rust conferences. 
+
+- **When**: February 19-20, 2025
+- **Where**: [The Brewery, 52 Chiswell Street, London, EC1Y 4SD, UK](https://maps.app.goo.gl/QakezC3gjkyo7sNr9)
+- **Format**: In-person workshops + conference
+- **Pricing**: 
+  - Conference Only: 304 dollars (early bird)
+  - Workshops starting at $300 + VAT
+- **CFP**: Closed
```
Comment (M3t0r) [inline content/blog/rust-conferences-2025/index.md:65]: CFP is rolled into "Links" for other entries. I like it seperated like this, but it should be consistent.

File: content/blog/rust-conferences-2025/index.md [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+So there you have it, our complete guide to all Rust developer conferences, workshops, and community meetups happening around the world in 2025. Whether you're interested in blockchain, embedded systems, or web development with Rust, there's an event for you.
+Let us know if we missed any events or if you have any updates to share. We'll see you at the next Rust conference!
+
+*Note: This list will be updated as more information becomes available. Some conferences are yet to announce their exact dates, venues, or ticket prices.*
+
+<script>
+// Add days until conference start
+function addDaysUntilConference() {
+    // Get all elements containing conference dates (assumes they're in <strong> tags after "When")
+    const dateElements = document.querySelectorAll('li strong:contains("When")');
```
Comment (M3t0r) [inline content/blog/rust-conferences-2025/index.md:268]: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/time

File: content/blog/rust-conferences-2025/index.md [hunk 9]
```diff
@@ -0,0 +1,258 @@
...
+    
+    dateElements.forEach(element => {
+        // Get the date text that follows the "When: " label
+        const dateText = element.nextSibling.textContent.trim();
```
Comment (M3t0r) [inline content/blog/rust-conferences-2025/index.md:272]: does this also handle the ": " that's outside of the `strong`?

File: content/blog/rust-conferences-2025/index.md [hunk 10]
```diff
@@ -0,0 +1,258 @@
...
+        
+        // Extract the start date (handles both single dates and ranges)
+        const startDate = dateText.split('-')[0].trim().split(',')[0];
+        
+        // Parse the date
+        const confDate = new Date(startDate + ", 2025");
```
Comment (M3t0r) [inline content/blog/rust-conferences-2025/index.md:278]: A bunch of conferences already have the ", 2025" suffix

File: content/blog/rust-conferences-2025/index.md [hunk 11]
```diff
@@ -0,0 +1,258 @@
...
+            
+            if (diffDays > 0) {
+                node.textContent = node.textContent.replace(
+                    dateText,
+                    `${dateText} (in ${diffDays} days)`
+                );
+            }
+        }
+    }
+}
```
Comment (M3t0r) [inline content/blog/rust-conferences-2025/index.md]: jQuery? What jQuery? The first implementation already looked native...

File: content/blog/rust-conferences-2025/index.md [hunk 12]
```diff
@@ -0,0 +1,258 @@
...
+
+// Run when the document is loaded
+document.addEventListener('DOMContentLoaded', addDaysUntilConference);
```
Comment (M3t0r) [inline content/blog/rust-conferences-2025/index.md]: You can also just call the function now. At the point where this script is parsed, the DOM is already present, and all the elements you want to modify have been loaded. Then there _really_ is no blink.

**Review by M3t0r — CHANGES_REQUESTED** [review]
Maybe you want to add the "Focus" field everywhere, even to "RustConf" with "Rust in general"?

File: content/blog/rust-conferences-2025/index.md (hunk 3 above)
Comment (mre) [reply content/blog/rust-conferences-2025/index.md:66]: done

File: content/blog/rust-conferences-2025/index.md (hunk 5 above)
Comment (mre) [reply content/blog/rust-conferences-2025/index.md]: oops, thanks

File: content/blog/rust-conferences-2025/index.md (hunk 7 above)
Comment (mre) [reply content/blog/rust-conferences-2025/index.md:65]: yeah, let's keep it separate

File: content/blog/rust-conferences-2025/index.md (hunk 2 above)
Comment (mre) [reply content/blog/rust-conferences-2025/index.md]: hehe, fixed

File: content/blog/rust-conferences-2025/index.md (hunk 6 above)
Comment (mre) [reply content/blog/rust-conferences-2025/index.md]: good idea

File: content/blog/rust-conferences-2025/index.md (hunk 9 above)
Comment (mre) [reply content/blog/rust-conferences-2025/index.md:272]: yeah ^^

File: content/blog/rust-conferences-2025/index.md (hunk 4 above)
Comment (mre) [reply content/blog/rust-conferences-2025/index.md]: removed

File: content/blog/rust-conferences-2025/index.md (hunk 8 above)
Comment (mre) [reply content/blog/rust-conferences-2025/index.md:268]: my js is not strong enough, haha

File: content/blog/rust-conferences-2025/index.md (hunk 10 above)
Comment (mre) [reply content/blog/rust-conferences-2025/index.md:278]: hm, it works though?

File: content/blog/rust-conferences-2025/index.md (hunk 11 above)
Comment (mre) [reply content/blog/rust-conferences-2025/index.md]: Yeah, copy-paste error. Sorry.

**Comment by mre** [comment]
Add all remaining info + banners.

File: content/blog/rust-conferences-2025/index.md (hunk 12 above)
Comment (mre) [reply content/blog/rust-conferences-2025/index.md]: done

**Comment by mre** [comment]
> Maybe you want to add the "Focus" field everywhere, even to "RustConf" with "Rust in general"?

Good idea. Added!

**Pull request merged**